app/error-navigation.js

```diff
@@ -0,0 +1,48 @@
+var dashify = require('./templates/dashify.js');
+var $ = require('jquery');
+
+var errorNavigation;
+
+function resetErrorNavigation(){
+  errorNavigation = {
+    index: 0,
+    errors: []
+  }
+};
+resetErrorNavigation();
+
+function registerErrors(elist){
+  errorNavigation.errors = elist;
+}
+
+function scrollToError(){
+    $("#"+dashify(errorNavigation.errors[errorNavigation.index].location))[0]
+    .scrollIntoView();
+}
+
+function moveError(by){
+  var i = errorNavigation.index;
+  var len = errorNavigation.errors.length;
+  var next = (((i + by) % len) + len) % len; // ensure positive, bounded
```

Comment:
;-) Yeah, me too (yesterday). And agreed, `init` would be much cleaner!

On Tue, Jun 21, 2016 at 9:49 AM, Erik Wiffin <notifications@github.com>
wrote:

> In app/error-navigation.js
> <https://github.com/sync-for-science/test-suite/pull/21#discussion_r67871212>
> :
>
> > +};
> > +resetErrorNavigation();
> > +
> > +function registerErrors(elist){
> > +  errorNavigation.errors = elist;
> > +}
> > +
> > +function scrollToError(){
> > +    $("#"+dashify(errorNavigation.errors[errorNavigation.index].location))[0]
> > +    .scrollIntoView();
> > +}
> > +
> > +function moveError(by){
> > +  var i = errorNavigation.index;
> > +  var len = errorNavigation.errors.length;
> > +  var next = (((i + by) % len) + len) % len; // ensure positive, bounded
>
> And I just learned something new and disappointing about javascript.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sync-for-science/test-suite/pull/21/files/209115eec5a159bb8d9484f1bf5ac63475c5d892#r67871212>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AATHAWeHzeF71We1SxsJcH_fIPa9-GtBks5qN-vwgaJpZM4I6Ibu>
> .
>